docs/docs/manual/grelfunctions.md

```diff
@@ -0,0 +1,537 @@
+
+If the omitFragment boolean is true, for example with `"internationalization".partition("nation", true)`, the fragment is not returned. The output is [ "inter", "alization" ].
+
+You can use regex for your fragment. The expresion `"abcdefgh".partition(/c.e/)` will output  [“abc”, "cde", defgh” ]. 
+
+###### rpartition(s, s or p fragment, b omitFragment (optional))
+
+Returns an array of strings [ a, fragment, z ] where a is the substring within s before the last occurrence of fragment, and z is the substring after the last instance of fragment. (Rpartition means “reverse partition.”) For example, `"parallel".rpartition("a")` returns 3 strings: [ par, a, llel ]. 
+
+Otherwise works identically to partition() above.
+
+### Encoding and hashing
+
+###### diff(s1 or d1, s2 or d2, s timeUnit (optional))
+
+Takes two strings or two dates and compares them, returning a string. The two objects must be the same data type. For strings, diff() returns the remainder of s2 starting with the first character where they differ. For example, `diff("cacti", "cactus")` returns "us". For dates, see [Date functions](#diffd1-d2-s-timeunit).
+
+###### escape(s, s mode)
+
+Escapes s in the given escaping mode. The mode can be one of: html, xml, csv, url, javascript. See the [recipes](https://github.com/OpenRefine/OpenRefine/wiki/Recipes#question-marks--showing-in-your-data) for examples of escaping and unescaping.
+
+###### unescape(s, s mode)
+
+Unescapes s in the given escaping mode. The mode can be one of: html, xml, csv, url, javascript. See the [recipes](https://github.com/OpenRefine/OpenRefine/wiki/Recipes#atampampt----att) for examples of escaping and unescaping. 
+
+###### md5(o)
+
+Returns the [MD5 hash](https://en.wikipedia.org/wiki/MD5) of an object. If fed something other than a string (array, number, date, etc.), md5() will convert it to a string and deliver the hash of the string. For example, `"[ and, or, not ]".md5()` will return 80fd34c2da7787a20c6c5e32e4899459.
+
+###### sha1(o)
+
+Returns the [SHA-1 hash](https://en.wikipedia.org/wiki/SHA-1) of an object. If fed something other than a string (array, number, date, etc.), sha1() will convert it to a string and deliver the hash of the string. For example, `"[ and, or, not ]".sha1()` will return a6664fc5476a043cabc179da1e3ce736e3959bbc.
+
+###### phonetic(s, s encoding)
+
+Returns a phonetic encoding of a string, based on an available phonetic algorithm. See the [section on phonetic clustering](cellediting#clustering-methods) for more information. Can be one of the following supported phonetic methods: [metaphone, doublemetaphone, metaphone3](https://www.wikipedia.org/wiki/Metaphone), [soundex](https://en.wikipedia.org/wiki/Soundex), [cologne](https://en.wikipedia.org/wiki/Cologne_phonetics). For example, “Ruth Prawer Jhabvala“.phonetic(metaphone3) outputs the string “R0PRRJPF”.
```

Comment:
We may wish to indicate the encoding algorithm needs to be in inverted commas

Reply:
Done, but will leave this open to remind myself in case we need to change some text based on #3330 .